test/assert.js

```diff
@@ -182,6 +182,110 @@ test('.throws()', function (t) {
 	t.end();
 });
 
+test('.throws() returning Error', function (t) {
+	t.doesNotThrow(function () {
+		var assertion = assert.throws(function () {
+			throw new Error('foo');
+		});
+
+		assert.same('foo', assertion.message);
```

Comment:
I'd prefer a strict identity check:

```js
var expected = new Error();
var actual = assert.throws(function () {
  throw expected;
});
t.equal(actual, expected);
```

Also note that you don't need the `t.doesNotThrow()` wrapper, just run a single `t.equal()` assertion. No need at all for the next test.